docs/USAGE-GUIDE.md

```diff
@@ -1585,6 +1585,102 @@ int s2n_config_add_ticket_crypto_key(struct s2n_config *config, const uint8_t *n
+
+**s2n_psk_set_secret** sets the out-of-band/externally provisioned pre-shared secret (PSK) for a given s2n_psk object. Note that the secret value is copied into s2n-tls memory and the user is responsbile for freeing the memory associated with the `secret` input. Be sure to call `free(secret)` after this API call.
+
+**s2n_psk_set_hmac** sets the PSK HMAC algorithm for a given PSK object.
+The supported pre-shared key HMAC algorithms are listed in the `s2n_psk_hmac` enum. Note that in versions >= TLS1.3 the cipher suite selected has the same Hash alogrithm as the pre-shared key. Use this API to override the default PSK Hash Algorithm value `S2N_PSK_HMAC_SHA256`. 
+
+Once the external PSK object is created and updated to include the PSK identity and secret, the caller can add the external PSK to the connection using `s2n_connection_append_psk` API. This API sets up the list of PSKs on the connection that will later be used by the pre-shared key extension to select a pre-shared key for the connection.
+
+**s2n_connection_append_psk** appends a pre-shared key object to the list of pre-shared keys supported by the s2n connection. If a pre-shared key with a duplicate identity is found, an error is returned and the pre-shared key is not added to the list. Note that the PSK object is copied to the connection and the user is responsbile for freeing the memory associated with the s2n_psk object. Use `s2n_psk_free` to free the memory associated with the external pre-shared key object after using this API.
+
+**s2n_config_set_psk_mode** sets the PSK mode on the s2n config object. The enum **s2n_psk_mode** lists the supported PSK modes. 
+
+**s2n_connection_set_psk_mode** sets the pre-shared key mode on the s2n connection object. The supported pre-shared key modes are listed in the enum `s2n_psk_mode`. This API overrides the pre-shared key mode for the connection and will continue to use the existing pre-shared key mode even when a new config is set for the connection. 
+
+### Obtaining Negotiated Pre-Shared Keys
+
+The folllowing APIs enable the caller to obtain the pre-shared key that was selected by the server for the connection. 
+
+**s2n_connection_get_negotiated_psk_identity_length** gets the negotiated pre-shared key identity length from the s2n connection object. If the negotiated pre-shared key does not exist, the value `0` is returned.  
+
+**s2n_connection_get_negotiated_psk_identity** gets the negotiated pre-shared key identity from the s2n connection object. If the negotiated pre-shared key does not exist, the PSK identity will not be obtained and no error will be returned.  
+
+### Configuring Client Offered Pre-Shared Keys
+ 
+The folllowing APIs enable the client to configure the list of pre-shared keys supported.
+
```

Comment:
These APIs are actually called by the server to process the client list, not by the client. They're not used to configure client PSKs.

Reply:
When I meant `configure the client offered pre-shared keys`  - I was referring to the list of PSKS sent from the client and processed by the server.